packages/manager/apps/pci-instances/src/pages/instances/instance/dashboard/hooks/useDashboardAction.ts

```diff
@@ -56,13 +64,98 @@ export const useUnattachedVolumes = ({
 
   return useMemo(
     () => ({
-      instance: instance?.name ?? '',
+      instance,
       volumes:
         instance?.volumes && volumes
-          ? selectUnattachedVolumes(volumes, instance.volumes)
+          ? selectUnattachedVolumes(
+              volumes,
+              instance.volumes,
+              instance.region.availabilityZone,
+            )
           : [],
       isPending: isInstancePending || isVolumePending,
     }),
     [instance, volumes, isInstancePending, isVolumePending],
   );
 };
+
+export const useAttachVolume = ({
+  projectId,
+  region,
+  instanceId,
+}: TUseUnattachedResource) => {
+  const { addError, addSuccess } = useNotifications();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const handleModalClose = () => navigate('..');
```

Comment:
issue (blocking): Navigation and notification should not be handle here

useAttachVolume should only handle the data/call part that are domain/technical things.
Notification and navigation depend on the caller and are pure front things.